runtime/memprof.c

```diff
@@ -334,264 +346,322 @@ struct tracked {
 #define Is_placeholder(block) \
   (Is_long(block) && (Long_val(block) & ~(uintnat)0xFFFF) == Placeholder_magic)
 
-/* When an entry is deleted, its index is replaced by that integer. */
-#define Invalid_index (~(uintnat)0)
+/* A resizable array of entries */
+struct entry_array {
+  struct tracked* t;
+  uintnat min_alloc_len, alloc_len, len;
+  /* Before this position, the [block] and [user_data] fields point to
+     the major heap ([young <= len]). */
+  uintnat young_idx;
+  /* There are no blocks to be deleted before this position
+     ([delete_idx <= entrieslen]). */
+  uintnat delete_idx;
+};
 
+#define MIN_ENTRIES_LOCAL_ALLOC_LEN 16
```

Comment:
The local and global arrays do not store the same kind of data: the local array stores entries waiting for the young alloc callback, so that it has a small number of entries (<< size of the minor heap multiplied by the sampling rate), and I don't expect that growing it is necessary in practice, while the global array contains all the rest, so that it may very well be much larger.

Reply:
May I mark this as resolved?